### PR TITLE
Simple linear regression - added explanations and code for how linear regression works

### PR DESCRIPTION
**Pull request by Harry-Leslie**

### CODE REVIEW

**Comment by Harry-Leslie** [comment]
# Simple Linear Models

Simple Linear Models are known as a linear model with only one feature and
one label. 

With simple linear regression you can use two main methods

## OLS (Ordinary Least Squares)

Ordinary Least Squares regression where you use simple optimisation methods (ie 
calculus) to find the coefficients

## Gradient Descent

Gradient descent is a common optimisation technique which reduces the error
which is similar to OLS however, gradient descent can work on a lot more
problems where the error is not easily differentiable which is the case with 
models like neural networks

I have written both the OLS and Gradient Descent implementation which are use to find the coefficients in the simple linear models
